soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -13044,7 +13107,13 @@ void func_8084DBC4(PlayState* play, Player* this, f32 arg2) {
 
     Player_GetMovementSpeedAndYaw(this, &sp2C, &sp2A, 0.0f, play);
     func_8084AEEC(this, &this->linearVelocity, sp2C * 0.5f, sp2A);
-    func_8084AEEC(this, &this->actor.velocity.y, arg2, this->currentYaw);
+    // #region SOH [Enhancement]
+    if (CVarGetInteger("gEnableWalkModify", 0)) {
+        SurfaceWithSwimMod(this, &this->actor.velocity.y, arg2, this->currentYaw);
```

Comment:
The implementation is kind of hard to follow, specifically with the copy of `func_8084AEEC` called `SurfaceWithSwimMod`, when in fact it's more like `SurfaceWithoutSwimMod` or `func_8084AEEC_original`. 

I think I understand the intention here, but maybe rename this function and leave a comment above this line explaining _why_ we want the original implementation of `func_8084AEEC` in this specific case when the swim mod is on.

Reply:
Intention with the original name was SurfaceWithSwimMod_Enabled_ - agree that the actual name didn't make any sense! Added a comment for the why here, and kept the note above the new function itself.

I also started with walk/swim unified, but built it out just in case - easier to have it built and cut it. I'm happy reunifying it to reduce bloat.